### DISCUSSION TITLE
Reading BLE advertisements in app layer

### DISCUSSION THREAD

**Discussion by saboulaban** [bitcraze/discussions#220] [Q&A]
Hi, I am new to crazyflie, what I am trying to achieve is to use bluetooth beacons for distance estimation. What I would like to achieve is to have my onboard app to:
- Access the RSSI received as received by nrf51822
- Payload of the BLE advertisement as received by nrf51822

My questions are:
1- Is this possible?
2- Can you point me to any documentation on how to achieve that?

Thank you.

**Comment by tobbeanton** [comment]
Well it depends on what you mean by possible. There is no API or similar ready for the payload of BLE advertisement, should be possible to implement though. The RSSI value is sent to the STM @100Hz. You can read it through the radio.rssi log value. Check out the[ RSSI ledring effect](https://github.com/bitcraze/crazyflie-firmware/blob/c6269c10c3abeea271718071e2d87a3babfb47f4/src/deck/drivers/src/ledring12.c#L759) for an example.
